pkg/linter/messages.yaml

```diff
@@ -4149,22 +4149,32 @@ LintCode:
       Future<void> createDir(String path) async {}
       ```
   do_not_use_environment:
-    problemMessage: "Invalid use of an environment declaration."
-    correctionMessage: "Try removing the environment declaration usage."
+    problemMessage: "Avoid using environment values like '{0}' which create hidden global state."
+    correctionMessage: "Try using 'Platform.environment' for runtime access or remove environment-dependent code."
     state:
       stable: "2.9"
     categories: [errorProne]
     hasPublishedDocs: false
     deprecatedDetails: |-
-      Using values derived from the environment at compile-time, creates
+      Using values derived from environment variables at compile-time creates
       hidden global state and makes applications hard to understand and maintain.
+      Environment values are resolved at compile-time and become embedded in the
+      compiled code, making behavior unpredictable across different environments.
```

Comment:
These constructors has nothing to do with "different environments" at all, they are, if anything, *predictable*.